### PR TITLE
Grianch and Pharaoh Hermits

### PR DESCRIPTION
**Pull request by alicetouhou**
Grianch and Pharaoh

I'm not the biggest fan of the Grianch portrait so I will probably change it. But I can't come up with something better atm. The pose is a bit weird but the biggest problem is his colors blend into the background, and I'm not sure what to do about that. Still I assume we have until the 24th to change the art so I'm not too worried.

### CODE REVIEW

File: common/cards/hermits/grianch_rare.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+					if (rowIndex === undefined) return 'FAILURE_INVALID_SLOT'
+					if (rowIndex === pickedPlayer.board.activeRow) return 'FAILURE_INVALID_SLOT'
+
+					if (pickResult.slot.type !== 'hermit') return 'FAILURE_INVALID_SLOT'
+					if (!pickResult.card) return 'FAILURE_INVALID_SLOT'
+
+					// Make sure it's an actual hermit card
+					const hermitCard = HERMIT_CARDS[pickResult.card.cardId]
+					if (!hermitCard) return 'FAILURE_INVALID_SLOT'
+					//C annot heal other pharaohs
```
Comment (Tyrannicodin) [inline common/cards/hermits/grianch_rare.ts]: This needs correction

File: common/config/debug-config.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
```
Comment (Tyrannicodin) [inline common/config/debug-config.json]: Debug config lol

File: common/cards/hermits/pharaoh-rare.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+					"Flip a coin. If heads, can give up to 80hp to AFK Hermit. Health given is equal to damage during attack. Can't heal other Pharaohs.",
+			},
+		})
+	}
+
+	override onAttach(game: GameModel, instance: string, pos: CardPosModel) {
+		const {player} = pos
+		const instanceKey = this.getInstanceKey(instance)
+
+		player.hooks.afterAttack.add(instance, (attack) => {
```
Comment (Tyrannicodin) [inline common/cards/hermits/pharaoh-rare.ts:37]: Might be wrong, but is this called every attack or only start all attacks have run

**Review by Tyrannicodin — CHANGES_REQUESTED** [review]

File: common/cards/hermits/grianch_rare.ts (hunk 1 above)
Comment (alicetouhou) [reply common/cards/hermits/grianch_rare.ts]: Shit did i forget to remove a few lines mb

File: common/config/debug-config.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
 	"forceCoinFlip": false,
 	"oneShotMode": false,
 	"disableDamage": false,
-	"disableDeckOut": false,
```
Comment (Tyrannicodin) [inline common/config/debug-config.json]: More debug config

**Review by Tyrannicodin — CHANGES_REQUESTED** [review]
One last debug config

File: common/config/debug-config.json (hunk 4 above)
Comment (alicetouhou) [reply common/config/debug-config.json]: RIP Finally fixed lol

File: common/cards/hermits/pharaoh-rare.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import {GameModel} from '../../models/game-model'
+import {getNonEmptyRows} from '../../utils/board'
+import {flipCoin} from '../../utils/coinFlips'
+import HermitCard from '../base/hermit-card'
+
+class PharaohRareHermitCard extends HermitCard {
+	constructor() {
+		super({
+			id: 'pharaoh_rare',
+			numericId: 173,
```
Comment (Tyrannicodin) [inline common/cards/hermits/pharaoh-rare.ts]: Should be 175
```suggestion
			numericId: 175,
```

**Review by Tyrannicodin — APPROVED** [review]

**Pull request merged**